src/components/Register/Register.js

```diff
@@ -1,75 +1,76 @@
 import React, { useState } from 'react';
+import FormUseWithValidation from '../FormUseWithValidation/FormUseWithValidation';
 import PopupWithForm from '../PopupWithform/PopupWithForm';
 import './Register.css';
 
-const Register = ({ isOpen, onClose, switchPopups }) => {
-  const [input, setInput] = useState({
-    email: '',
-    password: '',
-    username: '',
-  });
+const Register = ({ isOpen, onClose, switchPopups, onSubmit }) => {
+  const { handleChange, errors, isValid, handleSubmit } =
+    FormUseWithValidation(onSubmit);
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    console.log('submit reg');
-  };
-
-  const handleChange = (e) => {
-    const value = e.target.value;
-    setInput({
-      ...input,
-      [e.target.name]: value,
-    });
-    console.log('change reg input');
-  };
+  const [readOnly, setReadOnly] = useState(true);
 
   return (
     <PopupWithForm
       onClose={onClose}
+      isValid={isValid}
       onSubmit={handleSubmit}
       buttonText='Sign up'
       title='Sign up'
       name='register'
       isOpen={isOpen}
-      switchPopups={switchPopups}>
+      switchPopups={switchPopups}
+      errors={errors}>
       <label className='popup__input-label'>Email</label>
       <input
         className='popup__input'
         placeholder='Enter email'
+        type='email'
         name='email'
-        value={'' || input.email}
-        maxLength='30'
-        type='text'
+        readOnly={readOnly}
+        onFocus={() => setReadOnly(false)}
+        onBlur={() => setReadOnly(true)}
         onChange={handleChange}
         required
       />
+      <span
+        className={`popup__error ${
+          errors.email ? `popup__error_visible` : ''
+        }`}>
+        {errors.email}
+      </span>
       <label className='popup__input-label'>Password</label>
       <input
         className='popup__input'
         placeholder='Enter password'
+        type='password'
         name='password'
-        value={'' || input.password}
+        readOnly={readOnly}
+        onFocus={() => setReadOnly(false)}
+        onBlur={() => setReadOnly(true)}
         maxLength='30'
         minLength='8'
-        type='text'
         onChange={handleChange}

```

Comment:
If you have `onChange` or ` onInput` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.
To read more about **controlled components**: https://reactjs.org/docs/forms.html#controlled-components